docs/content/sips/008-using-oci-registries.md

```diff
@@ -0,0 +1,363 @@
+
+### Authentication
+
+Historically, the `docker` CLI has been the toolchain used to interact with
+container images and registries — as a result, given its popularity, the
+`spin oci` functionality should be able to re-use credentials for already
+logged-in users. Additionally, most container registry services have
+instructions on how to log in to their services using the `docker login`
+command.
+
+However, having Docker installed locally should not be a prerequisite for using
+Spin. To address this, a `spin oci login` command should be implemented that
+authenticates the Spin CLI to the desired registry instance:
+
+```bash
+$ spin oci login --username <username> --password <password>
+# OR
+$ echo $CONTAINER_REGISTRY_PASSWORD | spin oci login --username <username> --password-stdin
+```
+
+This user experience would mirror
+[the `docker login` command](https://docs.docker.com/engine/reference/commandline/login/).
+
+### Migrating applications from Bindle to an OCI registry
+
+Changing the distribution mechanism for Spin applications is a breaking change
+for the project — to address this, the project should provide functionality
+that users can use to migrate their applications from Bindle to an OCI registry.
+
+To take the ephemeral usefulness of this tool into account, and to prevent
+future breaking changes by the needing to remove it in the future,
+this functionality would be best suited as a Spin plugin, installed
+when needed and distributed separately:
+
+```bash
+spin bindle2oci \
+    --bindle-server <server> \
+    --bindle-username <username> \
+    --bindle-password <password> \
+    ---bindle-name <name> \
+    --oci <new-reference>
+```
+
+### Implementation
+
+A Spin application is made up of some metadata and component information
+together with the Wasm modules and static assets that made up those components.
+So conceptually, a Spin application is not *a single artifact*, but rather
+multiple distinct objects.
+
+This SIP proposes that, when distributed using an OCI registry, a Spin
+application would become a new OCI *artifact* with multiple layers
+(making the distinction clear, as *images* and *artifacts* are separate
+entities in OCI). Specifically, the media type used in this implementation
+for a Spin application is `application/vnd.fermyon.spin.application.v1+config`,
+and each Wasm module and static asset from the Spin components becomes an individual
+*layer* in the resulting registry entity. Because each file and Wasm module becomes
+a separate *layer*, we can efficiently de-duplicate and distribute applications.
+
+The remaining question is representing the Spin application definition — Spin
+introduced the internal representation of a *locked application* - an
+intermediate representation of a Spin application that *can* have a way to
+content-address Wasm modules and static assets. The implementation for this SIP
+uses the Spin locked application as the OCI configuration object.
```

Comment:
This effectively makes the locked app format part of the compatibility surface of spin. We should take a critical look at the format to see if there are any breaking changes we want to make before committing to that.